docs/source/graph_manipulation.rst

```diff
@@ -0,0 +1,61 @@
+.. _graph_manipulation:
+
+Advanced graph manipulation
+===========================
+These functions rewrite the dask graph underlying dask collections, producing equivalent
+collections with different sets of keys. They are intended to be used whenever the dask
+scheduler can't automatically figure out that the computation of a node should be
+delayed to avoid occupying RAM with its output for prolonged periods of time, and/or
+that recalculating a node is much cheaper than holding its output in memory.
+
+Consider the following example:
+
+.. code-block:: python
+
+   >>> import dask.array as da
+   >>> x = da.random.normal(size=5e9, chunks=1e7)
+   >>> x_avg = x.avg()
+   >>> y = (x - x_avg).topk(50000)[-1]
```

Comment:
I'll size it so that the unoptimized example runs in 4 GiB, so that a user can still observe the difference by glancing at his system monitor.
I had chosen ``topk(50000)[-1]`` as it produces a statistically stable output, wheresa max() is completely arbitrary. However I see your point.